### PR TITLE
Implements Slack notifications as an option

### PR DESCRIPTION
**Pull request by zedtux**
### Description

By setting the `default['fail2ban']['slack_webhook']` and `default['fail2ban']['slack_channel']` attributes, the recipe will add necessary stuff to get notifications in slack when an IP address is banned and unbanned.

Example of Slack notification message:
> [hostname] Banned 🇳🇬 217.117.13.12 in the jail sshd after 5 attempts

### Check List

- [x] All tests pass. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/TESTING.MD>
- [x] New functionality includes testing.
- [x] New functionality has been documented in the README if applicable
- [x] All commits have been signed for the Developer Certificate of Origin. See <https://github.com/chef-cookbooks/community_cookbook_documentation/blob/master/CONTRIBUTING.MD>

### CODE REVIEW

**Comment by zedtux** [comment]
Build failed due to a 503 error for Centos 7 when installing fail2ban package.

File: Gemfile [hunk 1]
```diff
@@ -1,3 +1,5 @@
+# frozen_string_literal: true
```
Comment (tas50) [inline Gemfile]: There's no need for this

File: Gemfile [hunk 2]
```diff
@@ -6,4 +8,6 @@
 
 source 'https://rubygems.org'
 
+gem 'berkshelf'
```
Comment (tas50) [inline Gemfile]: These need to come from ChefDK vs. managing deps in the Gemfile. We used to manage them here and it was a huge maint burden in all the repos

Comment (zedtux) [reply Gemfile]: Okay, no problem, it helped me to run the tests on my machine but I can revert it.

**Comment by zedtux** [comment]
Thank you for the review, let me fix those comments.

**Comment by zedtux** [comment]
Let me re-open a new PR with a clean branch

**Pull request closed without merging**